### DISCUSSION TITLE
How to load model from disk ?

### DISCUSSION THREAD

**Discussion by amitli1** [snakers4/silero-vad#334] [Q&A] [answered]
Im loading the **silero-vad** model using the following code:

```
   vad, vad_utils = torch.hub.load(repo_or_dir='snakers4/silero-vad',
                                    model='silero_vad',
                                    force_reload=False,
                                    onnx=False)
```

Once I did it, is it possible to load the model and it's utils from disk (and not from torch.hub) ?
How can I do it ?

**Comment by snakers4** [comment, marked as answer]
Hi,

Two simplest ways:

- Load the model via hub once, then use the cache flag in torch.hub.load;

- Clone the repo somewhere, then pass this path as a parameter to torch.hu.load, there should be a parameter like cache path or similar;

**Reply by snakers4** [reply]
Here - https://pytorch.org/docs/stable/hub.html#torch.hub.load

**Comment by mukundt** [comment]
@snakers4 does `torch.hub.load` load the model into CPU memory, or does it keep using the file on disk? I'm noticing that the VAD runs very slowly on a machine with an HDD, compared to an SSD.

**Reply by snakers4** [reply]
Torch hub loads model into RAM. Vad speed should not depend on disk speed, unless you write/ read from disk extensively.
